sdk/management/azure/src/main/java/com/azure/management/Azure.java

```diff
@@ -325,8 +331,25 @@ public Azure withSubscription(String subscriptionId) {
         @Override
         public Azure withDefaultSubscription() {
             if (profile.subscriptionId() == null) {
-                throw logger.logExceptionAsError(
-                    new IllegalArgumentException("Please specify the subscription ID for resource management."));
+                List<Subscription> subscriptions = new ArrayList<>();
+                this.subscriptions().list().forEach(subscription -> {
+                    subscriptions.add(subscription);
+                });
+                if (subscriptions.size() == 0) {
+                    throw logger.logExceptionAsError(
+                        new RuntimeException("Please create a subscription before you start resource management. "
```

Comment:
Is it possible to find a more specific exception type? RuntimeException is generally only viewed as abstract base class. Maybe IllegalStateException, or InvalidArgumentException (if we take that defaultSubscription only works for 1 subscription).

Reply:
Two candidates:
- `IllegalStateException` : 
> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.
- `UnsupportedOperationException` : 
> Thrown to indicate that the requested operation is not supported.

Maybe UnsupportedOperationException is better? The `withDefaultSubscription()` is not supported when no subscription found or more than one subscription found.

Reply:
`UnsupportedOperationException` is not good here. It is most commonly used in optional operation, which can be left unimplemented at all (like https://docs.oracle.com/javase/7/docs/api/java/util/Collection.html#add(E)).

So I would vote `IllegalStateException`.

Reply:
Got it. It's not the same as `InvalidOperationException` in csharp. So it's like the `withDefaultSubscription()` cannot be done at this time due to some restrictions.